esp/services/ws_store/ws_storeService.cpp

```diff
@@ -312,8 +312,23 @@ bool CwsstoreEx::onFetch(IEspContext &context, IEspFetchRequest &req, IEspFetchR
             storename = m_defaultStore.get();
     }
 
-    m_storeProvider->fetch(storename, req.getNamespace(), req.getKey(), value, secuser.get(), !req.getUserSpecific());
-    resp.setValue(value.str());
+    try
+    {
+        m_storeProvider->fetch(storename, req.getNamespace(), req.getKey(), value, secuser.get(), !req.getUserSpecific());
+        resp.setValue(value.str());
+    }
+    catch(IException * e)
+    {
+        if (e->errorCode() == ECLWATCH_INVALID_QUERY_KEY)
+        {
+            StringBuffer msg;
+            LOG(MCuserInfo, "WsStore: %s", e->errorMessage(msg).str());
```

Comment:
A bit concerned that this could lead to excessive logging (and potentially logging of sensitive information, depending on usage).

Reply:
That's definitely a valid concern which could apply to many logging statements. Perhaps it warrants adding a check-mark in the HPCC PR template to remind developers to be very mindful of the log output. and a full reassessment of logging levels and content.

As far as this particular change, suppressing the exception is intended to increase the log statement level from error to user-info. Depending on configuration it will appear less often now
I actually intended to output the offending key, and I will put that back in unless that in it self is one of your concerns.